### DISCUSSION TITLE
Update Content State API Not getting called on Completing HTML5 Content on portal

### DISCUSSION THREAD

**Discussion by paritshivani** [Sunbird-Ed/Community#754] [Bugs]
Hi team,
Hope you are doing well!

**Bug details:**
While consuming HTML content, the Update Content State API is expected to be triggered twice:
The first call occurs when the content consumption starts, with status=1.
The second call should happen upon completion of the content, with status=2.
However, the API is not getting triggered automatically on content completion. Instead, the API call with status=2 only occurs when the user clicks the replay button or maximizes the content player screen. This results in an incorrect update of the content state, as the completion status is not recorded properly unless the user takes these additional actions.
**Impact:**
Due to this issue, users will not be able to complete the course as their progress is not accurately tracked

@vinukumar-vs ,@rajeevsathish, @AmiableAnil , @Krishnaj20, @chitranshu-keshav, @SanthoshVasabhaktula, @gandham-santhosh , @coolbung @komalm @manishakapadnis

**Comment by Reshma-mahadik** [comment]
@vinukumar-vs ,@rajeevsathish, @AmiableAnil , @Krishnaj20, @chitranshu-keshav, @SanthoshVasabhaktula, @gandham-santhosh , @coolbung I would greatly appreciate any updates or insights you could provide on the matter.

**Comment by HarishGangula** [comment]
@paritshivani @Reshma-mahadik  Since we can't track the progress of HTML content directly and there's no way to detect when it ends, the player's end event, which internally updates the state, will be triggered upon navigating to the next content, changing routes, or replaying the content.
